src/generate-names.cc

```diff
@@ -104,30 +104,44 @@ class NameGenerator : public ExprVisitor::DelegateNop {
   Index num_memory_imports_ = 0;
   Index num_global_imports_ = 0;
   Index num_event_imports_ = 0;
+
+  bool numbered_names_;
+  bool dollar_prefix_;
 };
 
-NameGenerator::NameGenerator() : visitor_(this) {}
+NameGenerator::NameGenerator(bool numbered_names, bool dollar_prefix)
+  : visitor_(this), numbered_names_(numbered_names),
+    dollar_prefix_(dollar_prefix) {}
 
 // static
 bool NameGenerator::HasName(const std::string& str) {
   return !str.empty();
 }
 
-// static
 void NameGenerator::GenerateName(const char* prefix,
                                  Index index,
                                  unsigned disambiguator,
                                  std::string* str) {
-  *str = prefix;
+  if (dollar_prefix_) *str = "$";
```

Comment:
Kinda slick how this moves lots of hardcoded $'s from elsewhere.

Reply:
Actually I had first coded it to remove the `$` when necessary, then thought better of it :)